dotCMS/src/main/java/com/dotcms/variant/business/web/VariantWebAPIImpl.java

```diff
@@ -37,30 +38,73 @@ public class VariantWebAPIImpl implements VariantWebAPI{
      *
      * @return
      */
-    @Override
     public String currentVariantId() {
         final HttpServletRequest request = HttpServletRequestThreadLocal.INSTANCE.getRequest();
 
         if (!UtilMethods.isSet(request)) {
             return VariantAPI.DEFAULT_VARIANT.name();
         }
 
-        final String requestParameter = request.getParameter(VariantAPI.VARIANT_KEY);
+        String currentVariantName = getCurrentVariantNameFromParameter(request)
+                .or(() -> getCurrentVariantNameFromAttribute(request))
+                .orElse(null);
 
-        if (UtilMethods.isSet(requestParameter)) {
+        if (!UtilMethods.isSet(currentVariantName)) {
+            currentVariantName = VariantAPI.DEFAULT_VARIANT.name();
+        } else {
             try {
                 final Optional<Variant> byName = APILocator.getVariantAPI()
-                        .get(requestParameter);
-                return byName.isPresent() ? byName.get().name() : VariantAPI.DEFAULT_VARIANT.name();
+                        .get(currentVariantName);
+                currentVariantName = byName.isPresent() ? byName.get().name() : VariantAPI.DEFAULT_VARIANT.name();
             } catch (DotDataException e) {
                 Logger.error(VariantWebAPIImpl.class,
                         String.format("It is not possible get variant y name %s: %s",
-                                requestParameter, e.getMessage()));
-                return VariantAPI.DEFAULT_VARIANT.name();
+                                currentVariantName, e.getMessage()));
+                currentVariantName = VariantAPI.DEFAULT_VARIANT.name();
             }
-        } else {
-            return VariantAPI.DEFAULT_VARIANT.name();
         }
+
+        setSessionAttribute(request, currentVariantName);
+        return currentVariantName;
+    }
+
+    private static void setSessionAttribute(final HttpServletRequest request,
+            final String currentVariantName) {
+
+        final HttpSession session = request.getSession(true);
+
+        if (!UtilMethods.isSet(session)) {
```

Comment:
this does not make sense, since the true should always return a valid session, now may be you want to call getSession(false) to get the session only it was previously created